Tests/MapboxMapsTests/Foundation/MapboxMapTests.swift

```diff
@@ -219,4 +229,60 @@ final class MapboxMapTests: XCTestCase {
         projection = try? mapboxMap.mapProjection()
         XCTAssertEqual(projection, .globe())
     }
+
+    func testSubscribe() throws {
+        let observer = MockObserver()
+        let events: [String] = .random()
+        let mapboxObservable = try XCTUnwrap(mapboxObservableProviderStub.invocations.first?.returnValue as? MockMapboxObservable)
+
+        mapboxMap.subscribe(observer, events: events)
+
+        XCTAssertEqual(mapboxObservable.subscribeStub.invocations.count, 1)
+        XCTAssertIdentical(mapboxObservable.subscribeStub.invocations.first?.parameters.observer, observer)
+        XCTAssertEqual(mapboxObservable.subscribeStub.invocations.first?.parameters.events, events)
+    }
+
+    func testUnsubscribe() throws {
+        let observer = MockObserver()
+        let events: [String] = .random()
+        let mapboxObservable = try XCTUnwrap(mapboxObservableProviderStub.invocations.first?.returnValue as? MockMapboxObservable)
+
+        mapboxMap.unsubscribe(observer, events: events)
+
+        XCTAssertEqual(mapboxObservable.unsubscribeStub.invocations.count, 1)
+        XCTAssertIdentical(mapboxObservable.unsubscribeStub.invocations.first?.parameters.observer, observer)
+        XCTAssertEqual(mapboxObservable.unsubscribeStub.invocations.first?.parameters.events, events)
+    }
+
+    func testOnNext() throws {
+        let handlerStub = Stub<Event, Void>()
+        let eventType = MapEvents.EventKind.allCases.randomElement()!
+        let mapboxObservable = try XCTUnwrap(mapboxObservableProviderStub.invocations.first?.returnValue as? MockMapboxObservable)
+
+        mapboxMap.onNext(eventType, handler: handlerStub.call(with:))
+
+        XCTAssertEqual(mapboxObservable.onNextStub.invocations.count, 1)
+        XCTAssertEqual(mapboxObservable.onNextStub.invocations.first?.parameters.eventTypes, [eventType])
+        let handler = try XCTUnwrap(mapboxObservable.onNextStub.invocations.first?.parameters.handler)
+        let event = Event(type: "", data: 0)
+        handler(event)
+        XCTAssertEqual(handlerStub.invocations.count, 1)
+        XCTAssertIdentical(handlerStub.invocations.first?.parameters, event)
```

Comment:
This part looks more like testing out stubbing than testing `MapboxMap`, nuke it?

Reply:
The goal of this tests and the others like it is to verify that the handler that's passed to `MapboxMap.onNext` is effectively the same as the one passed to `MapboxObservable.onNext`. If Swift closures were identifiable, we could just compare the one received by MapboxObservable to the one passed to MapboxMap using `===`. Swift doesn't support that though, so instead, we verify that invoking the handler received by MapboxObservable causes the handler passed to MapboxMap to be invoked. This design also allows for the possibility that the handler passed to MapboxObservable is some wrapper around the one passed to MapboxMap. We don't take advantage of that possibility now, but if we did in the future, this test would continue to pass.

Reply:
I'm adding comments to explain this better

Reply:
It is clear now, thanks!